### PR TITLE
support separate file to output finalized tasks to

### PR DESCRIPTION
**Pull request by willscott**

### CODE REVIEW

**Review by raulk — COMMENTED** [review]
Can we write both ongoing events and finalized tasks, and identify each through a `type` field with values `ongoing_event`, `completed_task`?

This would allow us to track live dealbot activity and final data through ES/Kibana.

**Comment by willscott** [comment]
updates so the output struct is updated on stage transitions both for in progress and finalized tasks.

you can split on the top-level `status` field in the output json log lines, which will be `1` or `2` for `ongoing_event` and `3`(success) or `4`(failure) for `completed_task`'s

**Review by raulk — APPROVED** [review]
Can't fully say if the code is right, but if it does what it says on the tin, it should be in.

File: controller/state/statedb.go [hunk 1]
```diff
@@ -591,6 +615,29 @@ func (s *stateDB) Update(ctx context.Context, taskID string, req tasks.UpdateTas
 	return task, nil
 }
 
+func (s *stateDB) log(ctx context.Context, task tasks.Task, tx *sql.Tx) {
+	finalized, err := task.Finalize(ctx, txContextStorer(ctx, tx))
```
Comment (hannahhoward) [inline controller/state/statedb.go:619]: this is going to error if the task is not Successful / errored -- see https://github.com/filecoin-project/dealbot/blob/main/tasks/model.go#L348

is this what you want? It looks like you're trying to log all updates, but I think this is going to cause your code to skip updates that aren't to a success state.

**Review by hannahhoward — CHANGES_REQUESTED** [review]
Eek I think I see an issue but it's already merged -- see comment

**Pull request merged**